components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeServiceTest.java

```diff
@@ -79,260 +56,95 @@
-
-        assertNotNull(oAuth2ScopeService.updateScope(mockedScope), "Expected a not null object");
-    }
-
-    @Test(expectedExceptions = IdentityException.class)
-    public void testUpdateScopeWithNullName() throws Exception {
-        String name = null;
-        String description = "dummyScopeDescription";
-        mockedScope = new Scope(name, description);
-
-        oAuth2ScopeService.updateScope(mockedScope);
+        Scope dummyScope = new Scope(SCOPE_NAME, SCOPE_DESCRIPTION);
+        oAuth2ScopeService.registerScope(dummyScope);
+        Scope updatedDummyScope = new Scope(SCOPE_NAME, StringUtils.EMPTY);
+        assertEquals(oAuth2ScopeService.updateScope(updatedDummyScope).getDescription(), StringUtils.EMPTY);
+        oAuth2ScopeService.deleteScope(SCOPE_NAME);
     }
 
-    @Test(expectedExceptions = IdentityException.class)
-    public void testUpdateScopeWithScopeNotExists() throws Exception {
-        String name = "dummyScopeName";
-        String description = "dummyScopeDescription";
-        mockedScope = new Scope(name, description);
-
-        mockStatic(OAuthScopeCache.class);
-        when(OAuthScopeCache.getInstance()).thenReturn(mockedOAuthScopeCache);
-        when(mockedOAuthScopeCache.getValueFromCache(any(OAuthScopeCacheKey.class))).thenReturn(null);
-        when(mockedScopeMgtDAO.isScopeExists(anyString(), anyInt())).thenReturn(false);
-
-        oAuth2ScopeService.updateScope(mockedScope);
+    @Test(expectedExceptions = IdentityOAuth2ScopeException.class)
+    public void testUpdateScopeWithExceptions() throws Exception {
+        Scope updatedDummyScope = new Scope(SCOPE_NAME, StringUtils.EMPTY);
+        oAuth2ScopeService.updateScope(updatedDummyScope);
+        oAuth2ScopeService.deleteScope(SCOPE_NAME);
     }
 
-    @Test(expectedExceptions = IdentityOAuth2ScopeServerException.class)
-    public void testUpdateScopeWithIdentityOAuth2ScopeServerException() throws Exception {
-        String name = "dummyScopeName";
-        String description = "dummyScopeDescription";
-        mockedScope = new Scope(name, description);
-
-        mockStatic(OAuthScopeCache.class);
-        when(OAuthScopeCache.getInstance()).thenReturn(mockedOAuthScopeCache);
-        when(mockedOAuthScopeCache.getValueFromCache(any(OAuthScopeCacheKey.class))).thenReturn(mockedScope);
-        doThrow(new IdentityOAuth2ScopeServerException("dummyException")).when(mockedScopeMgtDAO).updateScopeByName
-                (any(Scope.class), anyInt());
-
-        oAuth2ScopeService.updateScope(mockedScope);
+    @Test(expectedExceptions = IdentityOAuth2ScopeException.class)
+    public void testDeleteScope() throws Exception {
+        Scope dummyScope = new Scope(SCOPE_NAME, SCOPE_DESCRIPTION);
+        oAuth2ScopeService.registerScope(dummyScope);
+        oAuth2ScopeService.deleteScope(SCOPE_NAME);
+        oAuth2ScopeService.getScope(SCOPE_NAME);
```

Comment:
We need to assert here for the value we get.

Reply:
If it throws an exception, no line get executed after line 148. So assertion would not be happened. If it doesn't throws an exception then it means a test failure. Even if we assert the output from getScope method, still we expect that test should throw an exception based on the input. So still test should fail regardless of output in assertion.